Library/Homebrew/formula.rb

```diff
@@ -595,6 +595,14 @@ def inspect
     s << ") #{path}>"
   end
 
+  def modified?
+    old_wd = Pathname.pwd
+    Dir.chdir path.dirname
+    diff = Utils.popen_read("git", "diff", "--diff-filter=M", "--name-only", "#{path}")
+    Dir.chdir old_wd
```

Comment:
You can do:
```ruby
path.parent.cd do
  diff = ...
end
```

Another thing that you need to check is formula file may be outside git repo.